contracts/test/StakeLocker.t.sol

```diff
@@ -280,11 +282,33 @@ contract StakeLockerTest is TestUtil {
         assertTrue(sam.try_transfer(address(stakeLocker), address(leo), 1 * WAD));
     }
 
+    function test_stake_transfer_lockup_period(uint256 stakeAmount) public {
+        uint256 start = block.timestamp;
+        stakeAmount = constrictToRange(stakeAmount, 1 wei, bPool.balanceOf(address(sam)), true);
+
+        makePublicAndStake(stakeAmount);
+
+        // Will fail because lockup period hasn't passed yet
+        assertTrue(!sam.try_transfer(address(stakeLocker), address(sid), stakeAmount));
+
+        // Warp to after lockup period and transfer
+        hevm.warp(start + stakeLocker.lockupPeriod());
```

Comment:
We should add another fail assert for `hevm.warp(start + stakeLocker.lockupPeriod() - 1);`